### ISSUE TITLE
feed_stage is counted on opposite sides by the CMO sweep and the MESH flow initialisation

### ISSUE THREAD

**Issue by jkitchin** [jkitchin/differentiable-flowsheets#212]
## Summary

`DistillationColumn` has two places that decide whether a stage is in the
rectifying or the stripping section, and they disagree about the feed stage
itself.

`src/difflow/units/distillation.py`, in `_solve_constant_molar_overflow`:

```python
def update_stage(x_above, j):
    is_rect = j >= p.feed_stage          # feed stage is RECTIFYING
```

and in `_solve_mesh`:

```python
# L[j] = liquid leaving stage j downward
L_init = jnp.where(jnp.arange(n) <= p.feed_stage, L_strip, L_rect)
                                         # feed stage is STRIPPING
```

Only one of these can be the intended convention.

## Which is right

For a saturated-liquid feed (`q=1`) entering stage `j = feed_stage`, the liquid
leaving that stage downward carries the feed, so `L[feed_stage] = L_rect + F =
L_strip` — the `_solve_mesh` line. The operating line used to *derive* the
composition at that stage is a balance over the envelope above it, which is all
rectifying — the `_solve_constant_molar_overflow` line. So the two may both be
locally defensible and the disagreement may be nominal rather than a bug. That
is exactly why it needs settling in one place rather than being decided twice.

## Impact

Bounded but not zero:

- In `_solve_mesh` the convention only sets `L_init`, and the energy-balance
  sweep recomputes `L`/`V` every iteration, so it washes out at convergence.
- In `_solve_constant_molar_overflow` it affects the answer, and that answer is
  returned directly when `use_mesh=False`.

Note that the CMO path currently appears insensitive to `feed_stage` altogether
(see #211), so the second bullet may be masked by a larger problem in the same
sweep. #211 should probably be settled first; this may fall out of it.

## Suggested fix

Pick one convention, write it down in `DistillationColumnParams.feed_stage`,
and use a single helper (`_is_rectifying(j)`) in both places so they cannot
drift again. A test that moves `feed_stage` across the boundary and checks the
section assignment on both solvers would pin it.

## Current state

`DistillationColumnParams.feed_stage` and `docs/unit-operations-chemical.md`
now describe the boundary stage as ambiguous rather than asserting a
convention, since asserting one would have documented a behaviour only half the
code has. That wording should be replaced once this is settled.

## Context

Found while correcting the distillation docs (#210 follow-up), which required
stating what `feed_stage` means.
